doc/langref.html.in

```diff
@@ -2180,16 +2180,20 @@ or
       Unlike normal structs, {#syntax#}packed{#endsyntax#} structs have guaranteed in-memory layout:
       </p>
       <ul>
-        <li>Fields remain in the order declared, least to most significant.</li>
+        <li>Fields are arranged in the order declared, from the least to the most significant bits of a backing integer.</li>
         <li>There is no padding between fields.</li>
-        <li>Zig supports arbitrary width {#link|Integers#} and although normally, integers with fewer
-        than 8 bits will still use 1 byte of memory, in packed structs, they use
-        exactly their bit width.
+        <li>The backing integer is subject to the same rules as any integer, including {#link|alignment|Alignment#}, 
+        having a maximum bit count of 65535, and the host endianness.
+        On a big endian system, the first declared field will have the highest memory address.
         </li>
+        <li>The backing integer has the same bit count as the fields' total bit width.</li> 
```

Comment:
In "The backing integer has the same bit count as the fields' total bit width", should this say "must have" instead of "has"?  (Also maybe add "If not inferred, ..."?)

Reply:
I don't think "must have" is best here because it communicates a constraint that the user must comply with, but the backing integer is constructed by the compiler for the user, so the user doesn't really need to do anything if they don't want to.

"Has" here just communicates a fact about the backing integer that is useful when communicating the memory layout.

Later in the doc, inferring vs. specifying the backing integer is explained as a readability and program correctness assertion feature, which I think is the correct way to explain it.

Reply:
updated inferred vs explicitly provided sentence

https://github.com/ziglang/zig/pull/21741/commits/af40751d5d10d7765c2062a685ff4a08516eaa70

to read:

> The backing integer can be inferred or explicitly provided. When inferred, it will be unsigned. When explicitly provided, its bit width will be enforced at compile time to exactly match the total bit width of the fields: